process/sync/baseSync.go

```diff
@@ -274,9 +274,9 @@ func (boot *baseBootstrap) ShouldSync() bool {
 
 	isNodeConnectedToTheNetwork := boot.networkWatcher.IsConnectedToTheNetwork()
 
-	shouldTrySync := boot.shouldTrySync()
+	boot.trySyncTrigger()
 
-	isNodeSynchronized := !boot.forkInfo.IsDetected && boot.hasLastBlock && isNodeConnectedToTheNetwork && !shouldTrySync
+	isNodeSynchronized := !boot.forkInfo.IsDetected && boot.hasLastBlock && isNodeConnectedToTheNetwork
```

Comment:
why is the condition deleted ? shouldTrySync ?

Reply:
shouldTrySync here would cause a change in the state of the node sync/not synced although this is not correct, as the only think it should do is request a higher nonce block, if available.

This has been changed to do exactly that (request the header only and if received it can then cause the sync).